code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -0,0 +1,761 @@
+			SSticker.news_report = NUKE_MISS
+		if(NUKE_RESULT_CREW_WIN_SYNDIES_DEAD)
+			SSticker.mode_result = "loss - evacuation - disk secured - syndi team dead"
+			SSticker.news_report = OPERATIVES_KILLED
+		if(NUKE_RESULT_CREW_WIN)
+			SSticker.mode_result = "loss - evacuation - disk secured"
+			SSticker.news_report = OPERATIVES_KILLED
+		if(NUKE_RESULT_DISK_LOST)
+			SSticker.mode_result = "halfwin - evacuation - disk not secured"
+			SSticker.news_report = OPERATIVE_SKIRMISH
+		if(NUKE_RESULT_DISK_STOLEN)
+			SSticker.mode_result = "halfwin - detonation averted"
+			SSticker.news_report = OPERATIVE_SKIRMISH
+		else
+			SSticker.mode_result = "halfwin - interrupted"
+			SSticker.news_report = OPERATIVE_SKIRMISH
+
+//////////////////////////////////////////////
+//                                          //
+//               REVS		                //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/roundstart/delayed/revs
+	name = "Revolution"
+	persistent = TRUE
+	antag_flag = ROLE_REV_HEAD
+	antag_datum = /datum/antagonist/rev/head
+	restricted_roles = list("AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Chief Medical Officer", "Research Director")
+	enemy_roles = list("AI", "Cyborg", "Security Officer", "Detective", "Head of Security", "Captain", "Warden")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 3
+	weight = 2
+	cost = 35
+	requirements = list(101,101,70,40,30,20,10,10,10,10)
+	high_population_requirement = 50
+	delay = 5 MINUTES
+	flags = HIGHLANDER_RULESET
+	var/required_heads = 3
+	var/datum/team/revolution/revolution
+	var/finished = 0
+
+/datum/dynamic_ruleset/roundstart/delayed/revs/ready(var/forced = 0)
+	if (forced)
+		required_heads = 1
+		required_candidates = 1
+	if (!..())
+		return FALSE
+	var/head_check = 0
+	for (var/mob/dead/new_player/player in GLOB.player_list)
+		if (player.mind.assigned_role in GLOB.command_positions)
+			head_check++
+	return (head_check >= required_heads)
+
+/datum/dynamic_ruleset/roundstart/delayed/revs/pre_execute()
+	var/max_canditates = 4
+	revolution = new()
+	for(var/i = 1 to max_canditates)
+		if(candidates.len <= 0)
+			break
+		var/mob/M = pick(candidates)
+		assigned += M
+		candidates -= M
+		M.mind.restricted_roles = restricted_roles
+		M.mind.special_role = antag_flag
+
+	return TRUE	
+
+/datum/dynamic_ruleset/roundstart/delayed/revs/execute()
+	for(var/mob/M in assigned)
+		var/datum/antagonist/rev/head/new_head = new antag_datum()
+		new_head.give_flash = TRUE
+		new_head.give_hud = TRUE
+		new_head.remove_clumsy = TRUE
+		M.mind.add_antag_datum(new_head,revolution)
+
+	revolution.update_objectives()
+	revolution.update_heads()
+	SSshuttle.registerHostileEnvironment(src)
+
+	return TRUE
+	
+/datum/dynamic_ruleset/roundstart/delayed/revs/rule_process()
```

Comment:
So fourth time writing this, lost all other three tries because I am writing this on my phone. The mode should have a check finished thing and check finished on highlander rulesets then call the round report from the ruleset that ended the round